cli/cliui/agent.go

```diff
@@ -381,6 +381,9 @@ func (d ConnDiags) Write(w io.Writer) {
 			_, _ = fmt.Fprintf(w, " - %s\n\n", msg)
 		}
 	}
+	if !d.PingP2P || d.Verbose || len(client) > 0 || len(agent) > 0 || len(general) > 1 {
```

Comment:
This should be something like `printTroubleShootingURL := false`, and set it to true in each place we print a warning

Reply:
That doesn't feel super great, you have to set it to true in ~12 different places. I'd much rather just check if we have at least one optional diagnostic.

Reply:
Well adding a condition to this if statement every time we add a new message to this output is also not good, so something probably needs to be changed to make this difficult for future changes to not forget to update this. You could wrap the print function in something that checks the flags instead for example

Reply:
I'm going with just checking the length of the lists to avoid needing to modify the condition later.